content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -13,11 +13,20 @@ export class _StartupOverlay extends React.PureComponent {
     this.removeOverlay = this.removeOverlay.bind(this);
     this.onInputInvalid = this.onInputInvalid.bind(this);
 
-    this.state = {emailInput: ""};
+    this.state = {emailInput: "", flowId: "", flowBeginTime: ""};
     this.initScene();
   }
 
   initScene() {
+    window.fetch(this.props.fxa_endpoint)
```

Comment:
This is actually not working as intended. If you log `this.props.fxa_endpoint` at this point, it's undefined. Therefore this actually throws all the time. You can put a log in the `.catch` and you can see that it throws:
`TypeError: "undefined is not a valid URL."`
The pref you're looking for is not ready in the constructor of this component unless you put it in the prerendered data's initial prefs values. The prefs get sent too late for this, and so you never get any of the data you're looking for, and it attempts to do a fetch on `undefined` which will throw

Reply:
I'm not getting that error, I was getting a cors error (since I accidentally lost the /metrics-flow portion of the URl when changing the contents of the pref.

Reply:
![screen shot 2018-07-04 at 11 07 13 am](https://user-images.githubusercontent.com/7219526/42284428-796be60a-7f7a-11e8-8ca3-613127d94ec8.png)